src/Compilers/Server/VBCSCompiler/CompilerRequestHandler.cs

```diff
@@ -98,50 +100,64 @@ public bool TryCreateCompiler(RunRequest request, BuildPaths buildPaths, [NotNul
+                Logger.Log($"Rejected: {request.RequestId}: {message}");
+                return new RejectedBuildResponse(message);
             }
 
             // Compiler server must be provided with a valid temporary directory in order to correctly
             // isolate signing between compilations.
             if (string.IsNullOrEmpty(request.TempDirectory))
             {
-                return new RejectedBuildResponse("Missing temp directory");
+                var message = "Missing temp directory";
+                Logger.Log($"Rejected: {request.RequestId}: {message}");
+                return new RejectedBuildResponse(message);
             }
 
             var buildPaths = new BuildPaths(ClientDirectory, request.WorkingDirectory, SdkDirectory, request.TempDirectory);
-            CommonCompiler? compiler;
-            if (!TryCreateCompiler(request, buildPaths, out compiler))
+            if (!TryCreateCompiler(request, buildPaths, out CommonCompiler? compiler))
             {
-                return new RejectedBuildResponse($"Cannot create compiler for language id {request.Language}");
+                var message = $"Cannot create compiler for language id {request.Language}";
+                Logger.Log($"Rejected: {request.RequestId}: {message}");
+                return new RejectedBuildResponse(message);
             }
 
             bool utf8output = compiler.Arguments.Utf8Output;
             if (!CheckAnalyzers(request.WorkingDirectory, compiler.Arguments.AnalyzerReferences, out List<string>? errorMessages))
             {
+                Logger.Log($"Rejected: {request.RequestId}: for analyer load issues {string.Join(";", errorMessages)}");
                 return new AnalyzerInconsistencyBuildResponse(new ReadOnlyCollection<string>(errorMessages));
             }
 
-            Logger.Log($"Begin {request.Language} compiler run");
-            TextWriter output = new StringWriter(CultureInfo.InvariantCulture);
-            int returnCode = compiler.Run(output, cancellationToken);
-            var outputString = output.ToString();
-            Logger.Log(@$"
-End {request.Language} Compilation complete.
+            Logger.Log($"Begin {request.RequestId} {request.Language} compiler run");
+            try
+            {
+                TextWriter output = new StringWriter(CultureInfo.InvariantCulture);
+                int returnCode = compiler.Run(output, cancellationToken);
+                var outputString = output.ToString();
+                Logger.Log(@$"End {request.RequestId} {request.Language} compiler run
 Return code: {returnCode}
 Output:
 {outputString}");
-            return new CompletedBuildResponse(returnCode, utf8output, outputString);
+                return new CompletedBuildResponse(returnCode, utf8output, outputString);
+            }
+            catch (Exception ex)
+            {
+                Logger.LogException(ex, $"Running compilation for {request.RequestId}");
```

Comment:
Is the message correct?

Reply:
Yes. The second argument to `LogException` describes the operation which was occurring when the exception was thrown.